ios/FluentUI/DrawerVnext/Drawer.swift

```diff
@@ -0,0 +1,224 @@
+ */
+public struct Drawer<Content: View>: View {
+
+    @ObservedObject var state: DrawerState
+    @ObservedObject var tokens: DrawerTokens
+
+    var backgroundLayerColor: Color! = Color.black
+    private var backgroundLayerOpacity: Double! {
+        return state.backgroundDimmed ? 0.5 : 0
+    }
+    var backgroundLayerAnimation: Animation! = Animation.default
+
+    var percentWidthOfContent: CGFloat! = 0.9
+
+    // content view on top of
+    public var content: Content
+
+    // `ivar` to keep track of dragged offset
+    @State private var draggedOffsetWidth: CGFloat?
+    private func computedOffset(screenSize: CGSize) -> CGFloat {
+
+        if let draggedOffsetWidth = draggedOffsetWidth {
+            return draggedOffsetWidth
+        }
+
+        var width = CGFloat.zero
+        let contentWidth = screenSize.width * percentWidthOfContent
+        if !state.isExpanded { // if closed then hidden behind the screen
+            width = state.presentationDirection == .left ? -contentWidth : contentWidth
+        }
+
+        return width
+    }
+
+//    // percentage of space occupied by drawer when open
+//    private var openDrawerWidth: CGFloat {
+//        return UIScreen.main.bounds.width * 0.9
+//    }
+
+    public init(content: Content) {
+        self.content = content
+        tokens = DrawerTokens()
+        state = DrawerState()
+    }
+
+    public var body: some View {
+
+        GeometryReader { reader in
+            HStack {
+
+                if state.presentationDirection == .right {
+                    Spacer()
+                }
+
+                content
+                    .frame(width: reader.size.width * percentWidthOfContent)
+                    .shadow(color: tokens.shadowColor.opacity(state.isExpanded ? tokens.shadowOpacity : 0),
+                            radius: tokens.shadowBlur,
+                            x: tokens.shadowDepth[0],
+                            y: tokens.shadowDepth[1])
+                    .offset(x: computedOffset(screenSize: reader.size))
+                    .animation(backgroundLayerAnimation)
+
+                if state.presentationDirection == .left {
+                    Spacer()
+                }
+
+            }.onTapGesture {
+                state.isExpanded.toggle()
+            }.background(state.isExpanded ? backgroundLayerColor.opacity(backgroundLayerOpacity) : Color.clear)
+            .gesture(drag)
+        }
+    }
+
+    var drag: some Gesture {
+        DragGesture()
+            .onChanged { value in
+
+                let withinDragBounds = state.presentationDirection == .left ? value.translation.width < 0 : value.translation.width > 0
+                if withinDragBounds {
+                    draggedOffsetWidth = value.translation.width
+                }
+            }
+            .onEnded { _ in
+                if let draggedOffsetWidth = draggedOffsetWidth {
+                    if abs(draggedOffsetWidth) < 270 / 4 {
```

Comment:
Where is 270 and 4 coming from?
Shouldn't this be relative to the width?

Reply:
Thanks for catching this, I have updated this to use geometry reader instead